### ISSUE TITLE
Use a priority queue as opposed to min_elt from Map

### ISSUE THREAD

**Issue by rleonid** [hammerlab/prohlatype#48]
Remove this abuse.

**Comment by rleonid** [comment]
I've tried several attempts to use a smarter data structure:


1.  [a priority queue and hash table combo](https://github.com/hammerlab/prohlatype/tree/use_prioqueue_for_align).  This turned out to be a bit worse. My unverified suspicion is that the cost of allocating the hashtable, and queue is much greater than just the map. 
2. [a queue of lists](https://github.com/hammerlab/prohlatype/tree/hand_craft_prio_queue). This    unfortunately has O(n) insert cost, but on average it would insert at almost the front so it shouldn't be that bad (1,2 comparisons to find the right insert point). But it didn't outperform the status-quo on 2 benchmarks. I tried this to compare against the weight of the hash table. And it was closer in comparison to the status quo.

In all of the tests these approaches were only incrementally, almost indiscernibly, slower. My suspicion was that this wasn't the biggest bottleneck to address next. Actually running the code through the profiler also points to the `O(k)` assigning cost problem (the fold over the entire bitset).

Consequently, this issue will not be addressed by the [faster](https://github.com/hammerlab/prohlatype/pull/68) branch, and the issue is going on to the backlog until I think it is worth while.

I did not experiment with removing the uses of `min_elt` during graph construction, only alignment. But that is also of lower priority (no pun intended).
